golang/explode.go

```diff
@@ -0,0 +1,27 @@
+package main
+
+import (
+	"errors"
+	"strings"
+)
+
+func Explode(separator string, str string, limit ...int) []string {
+	if separator == "" {
+		return []string{errors.New("Separator cannot be empty").Error()}
+	}
+
+	var result []string
+	result = strings.Split(str, separator)
+
+	if limit == nil || limit[0] > len(result) || limit[0] < -len(result) {
```

Comment:
Ada kemungkinan length dari slice limit itu 0, nanti kena panic index out of bounds. Harus di guard dulu.

Reply:
noted

Reply:
nol kosong